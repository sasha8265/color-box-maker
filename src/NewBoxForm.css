.NewBoxForm {
    margin-bottom: 30px;
}


.NewBoxForm-inputGroup {
    margin-right: 20px;
    margin-left: 5px;
}


/* button */