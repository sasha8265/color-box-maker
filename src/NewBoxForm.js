import React, { useState } from "react";

/** Form for creating a new colored box to add to the page.
 *
 * Has state for the height/width and color of the item; 
 * on submission, sends {height, width, color} to fn rec'd from parent.
 * 
 */

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = { height: "", width: "", color: "" }
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    /**Send {height, width, color} to parent and clear form */

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({ ...formData });
        setFormData(INITIAL_STATE);
    }

    return (
        <form className="NewBoxForm" onSubmit={handleSubmit}>
            <label htmlFor="height" >Height </label>
            <input
                id="height"
                type="text"
                name="height"
                placeholder="100"
                value={formData.height}
                onChange={handleChange} />
            
            <label htmlFor="height" >Width </label>
            <input
                id="width"
                type="text"
                name="width"
                placeholder="100"
                value={formData.width}
                onChange={handleChange} />
            
            <label htmlFor="color" >Color </label>
            <input
                id="color"
                type="color"
                name="color"
                // placeholder="100"
                value={formData.color}
                onChange={handleChange} />
            
            <button>Add</button>
        </form>
    )

}

export default NewBoxForm;