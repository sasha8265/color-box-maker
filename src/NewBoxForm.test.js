import React from "react";
import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm"

//smoke test
it("renders without crashing", function () {
    render(<NewBoxForm />);
});

//snapshot test
it("matches snapshot", function () {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});