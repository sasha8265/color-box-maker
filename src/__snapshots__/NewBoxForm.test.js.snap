// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <span
      class="NewBoxForm-inputGroup"
    >
      <label
        for="height"
      >
        Height 
      </label>
      <input
        id="height"
        name="height"
        placeholder="100"
        type="text"
        value=""
      />
    </span>
    <span
      class="NewBoxForm-inputGroup"
    >
      <label
        for="height"
      >
        Width 
      </label>
      <input
        id="width"
        name="width"
        placeholder="100"
        type="text"
        value=""
      />
    </span>
    <span
      class="NewBoxForm-inputGroup"
    >
      <label
        for="color"
      >
        Color 
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value=""
      />
    </span>
    <span
      class="NewBoxForm-button"
    >
      <button>
        Add
      </button>
    </span>
  </form>
</DocumentFragment>
`;
